### PR TITLE
Switch CC installation test from autoyast to libyui

### PR DESCRIPTION
**Pull request by paolostivanin**
Ticket: https://progress.opensuse.org/issues/156655
VRs:
- 15-SP6: https://openqa.suse.de/tests/overview?distri=sle&version=15-SP6&build=77.1&groupid=431
- 15-SP5-QR: https://openqa.suse.de/tests/overview?distri=sle&version=15-SP5&build=137.7&groupid=431

Please note that s390x is green only because I've applied a workaround to the current known ssh bug. Installation tests will be red on that platform until that bug is fixed.

### CODE REVIEW

**Comment by paolostivanin** [comment]
Could you (@tjyrinki , @jknphy) please review this PR? Thanks

File: schedule/yast/sle/flows/disable_beta.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+# In QU/maintenance we disable the beta stuff
+access_beta: []
```
Comment (jknphy) [inline schedule/yast/sle/flows/disable_beta.yaml:3]: we are going to remove this feature, it is just a prototype. Only one overwrite with default schedule is maintainable.

File: tests/installation/common_criteria_configuration/cc_disk_encryption_password.pm [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# Summary: The test module configure the disk encryption password
+# when using the Common Criteria role.
+# Maintainer: QE Security <none@suse.de>
+
+use parent 'y2_installbase';
+use strict;
+use warnings;
+
+sub run {
+    my $partitioning_scheme = $testapi::distri->get_common_criteria_configuration();
```
Comment (jknphy) [inline tests/installation/common_criteria_configuration/cc_disk_encryption_password.pm:13]: the name of the variable should correspond with the return of the function.

File: tests/installation/common_criteria_configuration/cc_disk_encryption_password.pm [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Copyright SUSE LLC
```
Comment (jknphy) [inline tests/installation/common_criteria_configuration/cc_disk_encryption_password.pm:1]: cc is not well known by others, it would better to use "common_criteria" and a shorten a bit the rest.

File: schedule/security/flows/default_aarch64.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
+---
```
Comment (jknphy) [inline schedule/security/flows/default_aarch64.yaml:1]: why did you need to create new default schedules? we don't plan to change those much, should be save to share among squad I think (if doesn't work, can be also duplicated later), did you find any keys that were different compare with what we have in maintenance? maybe it is the case as we don't do quarter updates but just to see the difference.

**Comment by jknphy** [comment]
Look good in general, good job! if you can wait a bit to next week, I can give it a more deep review (I"m not much available this one).

File: schedule/yast/sle/flows/disable_beta.yaml (hunk 1 above)
Comment (paolostivanin) [reply schedule/yast/sle/flows/disable_beta.yaml:3]: ah sorry, forgot to remove that file (was a test, I'm not using it)

File: tests/installation/common_criteria_configuration/cc_disk_encryption_password.pm (hunk 2 above)
Comment (paolostivanin) [reply tests/installation/common_criteria_configuration/cc_disk_encryption_password.pm:13]: done

File: schedule/security/flows/default_aarch64.yaml (hunk 4 above)
Comment (paolostivanin) [reply schedule/security/flows/default_aarch64.yaml]: I've cleaned it up and move to your defaults. Let's see how it will go with VRs :)

Comment (jknphy) [reply schedule/security/flows/default_aarch64.yaml]: the trick is to spread the complexity among the different solutions, you cannot get rid of it, so the yaml needs be a bit complex (with one single 1 default file, not many or having conditional schedule) the actions of testing can be done using the distribution pattern for example (it is a bit complex but not much) and the test data to validate can be a bit complex (having different files in yaml with test data depending on each arch for example) but not too complex (containing many if/elses).

**Comment by paolostivanin** [comment]
@jknphy we don't have working CC tests without this, but since no new build is scheduled for now I guess we can wait a bit more :smile:

**Review by tjyrinki — APPROVED** [review]
It looks good and results are great, but obviously Joaquin is the libyui expert here.

File: schedule/yast/sle/flows/install_sles.yaml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# Used by QU aarch64
+product_selection:
+  - installation/product_selection/install_SLES
```
Comment (jknphy) [inline schedule/yast/sle/flows/install_sles.yaml:3]: this file you forgot to remove it.

File: tests/installation/boot_encrypt.pm [hunk 6]
```diff
@@ -12,9 +12,21 @@ use strict;
 use warnings;
 use base "installbasetest";
 use utils;
+use version_utils qw(is_sle);
+use Utils::Architectures qw(is_aarch64);
+use testapi;
 
 sub run {
-    unlock_if_encrypted(check_typed_password => 1) if need_unlock_after_bootloader;
+    if (is_aarch64 && is_sle('<=15-SP5') && get_var('FLAVOR', 'Online-QR') && get_var('ENCRYPT', '1') && get_var('FULL_LVM_ENCRYPT', '1')) {
```
Comment (jknphy) [inline tests/installation/boot_encrypt.pm:20]: As this module is owned by Yam squad and we don't test in QU (for now), I would strongly suggest here that you create your own test module as was at the beginning, we don't want to maintain this mysterious conditions, it is hard to read and it will only be worse with time, the code will degenerate with other contributors adding something else.
I believe this could have been done inside `unlock_if_encrypted` as I discussed with you offline (needs to tried), that would be better solution, perhaps for a future ticket in your squad. You could just copy/paste to `boot_encrypt_no_video_present` and have different schedules, once that future ticket is solve you can unify it, but it is up to you (in our experience you can only reduce schedule file with multiple refactors).

File: tests/installation/common_criteria_configuration/common_criteria_encryption_pwd.pm [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+# Summary: The test module configure the disk encryption password
+# when using the Common Criteria role.
+# Maintainer: QE Security <none@suse.de>
+
+use parent 'y2_installbase';
+use strict;
+use warnings;
+
+sub run {
+    my $common_criteria_configuration_page = $testapi::distri->get_common_criteria_configuration();
```
Comment (jknphy) [inline tests/installation/common_criteria_configuration/common_criteria_encryption_pwd.pm:13]: just `$common_criteria_configuration`, but not page, you are using the "controller" not the "page".

File: tests/security/atsec/setup_atsec_env.pm [hunk 8]
```diff
@@ -19,7 +19,7 @@ sub run {
 
     select_console 'root-console';
 
-    zypper_call('in wget gcc make');
+    zypper_call('in wget gcc make curl');
```
Comment (jknphy) [inline tests/security/atsec/setup_atsec_env.pm:22]: you need to squash all your commit (I saw 2 debug ones I think) all in this PR seems related with the UI, except this one change that I would recommend to put it in a different commit, make sense? (just in case something need to be reverted, we hope not! :) you don't need to revert stuff that is fine).

**Comment by jknphy** [comment]
All very neat, only few modifications requested.

File: schedule/yast/sle/flows/install_sles.yaml (hunk 5 above)
Comment (paolostivanin) [reply schedule/yast/sle/flows/install_sles.yaml:3]: ah right, thanks

File: tests/installation/common_criteria_configuration/common_criteria_encryption_pwd.pm (hunk 7 above)
Comment (paolostivanin) [reply tests/installation/common_criteria_configuration/common_criteria_encryption_pwd.pm:13]: done

File: tests/security/atsec/setup_atsec_env.pm (hunk 8 above)
Comment (paolostivanin) [reply tests/security/atsec/setup_atsec_env.pm:22]: we can keep it together. This is actually needed because of this new installation method being used. With autoyast it wasn't required since curl was already installed.

File: tests/installation/boot_encrypt.pm (hunk 6 above)
Comment (paolostivanin) [reply tests/installation/boot_encrypt.pm:20]: done

File: tests/security/boot_encrypt_no_video_present.pm [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+# Summary: Unlock encrypted partitions during bootup after the bootloader
+#   passed, e.g. from plymouth.
+# Maintainer: QE Security <none@suse.de>
+
+use strict;
+use warnings;
+use base "installbasetest";
+use utils;
+use version_utils qw(is_sle);
+use Utils::Architectures qw(is_aarch64);
```
Comment (jknphy) [inline tests/security/boot_encrypt_no_video_present.pm:14]: check unused imports.

**Review by jknphy — APPROVED** [review]
